docs/linux/fs.md

```diff
@@ -592,38 +594,41 @@ must be specified.
 
 ## Btrfs
 
-*Btrfs* is a file system based on the copy-on-write (CoW) principle resulting
-in any block update to never be written in-place. *Btrfs* currently supports
-zoned block devices, but that support is experimental.
+*Btrfs* is a file system based on the copy-on-write (CoW) principle. This
+principle has the result that no block update can be written in-place. 
+*Btrfs* currently supports zoned block devices, but that support is 
+experimental.
 
 ### Zoned Block Device Support
 
-Zoned block device support was added to *btrfs* with kernel 5.12. Since
+Zoned block device support was added to *btrfs* with kernel 5.12. Because 
 super-blocks are the only on-disk data structure with a fixed location in
 *btrfs*, zoned block device support introduces the concept of log-structured
-super-blocks. For each of the super-blocks (primary and backup super-blocks)
-in *btrfs*, zoned mode reserves two consecutive zones to hold these
-super-blocks. When a new super-block is being written, it is appended to the
-respective super-block zone. Once the first super-block zone is filled, the
-next super block is being written to the second super-block zone and the first
-is being reset. The currently valid and active super-block can be determined
-by looking at the position of the zone write pointer of the respective
-super-block zones.
+super-blocks (TODO: explain how long-structured super-blocks relate to
+super-blocks in btrfs). Zoned mode reserves two consecutive zones to hold each 
+of the super-blocks (primary and backup super-blocks) in *btrfs*. When a new 
+super-block is written, it is appended to its respective super-block 
+zone. After the first super-block zone is filled, the next super block is 
+written to the second super-block zone and the first is reset. The super-block 
+that is currently valid and active can be determined by looking at (TODO:
+explain what is looking at the position of the zone write pointer) the 
```

Comment:
looking = inspecting (testing the value) the zone write pointer position

Reply:
I understand that. What I don't understand is what is doing the looking. Which thing inspects the zone write pointer position?

Reply:
The mount process is doing the "looking" to find the latest version (most up to date) super block among all the blocks written in the log zones. To clarify, the sentence:

The super-block that is currently valid and active can be determined by looking at position of the zone write pointer of the respective super-block zones.

could be replaced with something like this:

At mount time, btrfs can find the latest version of the super-block by looking at position of the zone write pointer of the respective super-block zones. The most recent and valid super-block is always the last block stored before the write pointer position.

Hmm ?

Reply:
Yes. This is excellent.

I'll push this back to you by the morning.

Reply:
I got it done a little sooner than I expected.

Back at you, Damien.